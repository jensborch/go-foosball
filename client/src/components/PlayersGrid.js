import React from 'react';
import PropTypes from 'prop-types';
import Card from '@material-ui/core/Card';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';

class Player extends React.Component {
  state = { ranking: this.props.ranking };

  select = () => {
    this.props.select(
      this.props.tournament,
      this.props.player.nickname,
      parseInt(this.state.ranking, 10)
    );
  };

  render() {
    const { classes, player } = this.props;
    return (
      <Card className={classes.card} key={player.nickname}>
        <div className={classes.cell} />
        <CardContent>
          <Typography gutterBottom variant="h5" component="h3">
            {player.nickname} - {player.realname}
          </Typography>
        </CardContent>
        <CardActions>
          <div>
            <TextField
              type="number"
              value={this.state.ranking}
              onChange={(e) => this.setState({ ranking: e.target.value })}
              helperText="Ranking"
              lable="Ranking"
              margin="dense"
            />
            <Button
              variant="outlined"
              className={classes.cardButton}
              onClick={this.select}
            >
              Add
            </Button>
          </div>
        </CardActions>
      </Card>
    );
  }
}

class NewPlayer extends React.Component {
  state = {
    nickname: '',
    realname: '',
  };

  create = () => {
    this.props.create(this.state.nickname, this.state.realname);
  };

  render() {
    const { classes } = this.props;
    return (
      <Card className={classes.card}>
        <CardContent>
          <Grid container direction="column">
            <Grid item>
              <TextField
                helperText="Name"
                value={this.state.realname}
                onChange={(event) =>
                  this.setState({
                    realname: event.target.value,
                  })
                }
                lable="Name"
              />
            </Grid>
            <Grid item>
              <TextField
                helperText="Nickname"
                value={this.state.nickname}
                onChange={(event) =>
                  this.setState({ nickname: event.target.value })
                }
                lable="Nickname"
              />
            </Grid>
          </Grid>
        </CardContent>
        <CardActions>
          <Button
            variant="outlined"
            className={classes.cardButton}
            onClick={this.create}
          >
            Create
          </Button>
        </CardActions>
      </Card>
    );
  }
}

class PlayersGrid extends React.Component {
  componentDidMount() {
    if (this.props.fetch) {
      this.props.fetch(this.props.id);
    }
  }

  render() {
    const { classes, players } = this.props;
    return (
      <Grid container spacing={16} direction="row">
        {players.map((p, i) => (
          <Grid item key={p.nickname}>
            <Player
              player={p}
              tournament={this.props.tournament}
              ranking={this.props.ranking}
              select={this.props.select}
              classes={classes}
            />
          </Grid>
        ))}
        <Grid item>
          <NewPlayer classes={classes} create={this.props.create} />
        </Grid>
      </Grid>
    );
  }
}

PlayersGrid.propTypes = {
  classes: PropTypes.object.isRequired,
  players: PropTypes.array.isRequired,
  tournament: PropTypes.string.isRequired,
  ranking: PropTypes.number.isRequired,
  fetch: PropTypes.func,
  create: PropTypes.func.isRequired,
};

NewPlayer.propTypes = {
  classes: PropTypes.object.isRequired,
  create: PropTypes.func.isRequired,
};

Player.propTypes = {
  classes: PropTypes.object.isRequired,
  player: PropTypes.object.isRequired,
  tournament: PropTypes.string.isRequired,
  ranking: PropTypes.number.isRequired,
  select: PropTypes.func.isRequired,
};

export default PlayersGrid;
