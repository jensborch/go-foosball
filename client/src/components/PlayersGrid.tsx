import * as Api from '../api/Api';
import { useState } from 'react';
import {
  Card,
  CardContent,
  Grid,
  Typography,
  Button,
  TextField,
  CardActions,
  Box,
  CircularProgress,
} from '@mui/material';
import { Error } from './Error';
import {
  usePlayerMutation,
  useTournamentPlayerMutation,
  useTournamentPlayers,
} from '../api/hooks';

type PlayerProps = {
  tournament: string;
  player: Api.TournamentPlayer;
};

const Player = (props: PlayerProps) => {
  const [player, setPlayer] = useState(props.player);
  const { mutate } = useTournamentPlayerMutation(props.tournament);

  const onAddPlayer = () => {
    mutate({
      nickname: player.nickname,
      ranking: player.ranking,
    });
  };

  return (
    <Card
      sx={{
        height: '250px',
        margin: (theme) => theme.spacing(3),
      }}
      key={player.nickname}
    >
      <Box
        sx={{
          height: '100px',
          backgroundColor: (theme) => theme.palette.grey[300],
        }}
      />
      <CardContent>
        <Typography gutterBottom variant="h5" component="h3">
          {player.nickname} - {player.realname}
        </Typography>
      </CardContent>
      <CardActions>
        <div>
          <TextField
            type="number"
            value={player.ranking}
            onChange={(e) =>
              setPlayer({
                ...player,
                ranking: parseInt(e.target.value),
              })
            }
            helperText="Ranking"
            label="Ranking"
            margin="dense"
          />
          <Button
            sx={{
              margin: (theme) => theme.spacing(),
            }}
            variant="outlined"
            onClick={onAddPlayer}
          >
            Add
          </Button>
        </div>
      </CardActions>
    </Card>
  );
};

const NewPlayer = () => {
  const [nickname, setNickname] = useState('');
  const [realname, setRealname] = useState('');
  const { mutate } = usePlayerMutation();

  const onCreatePlayer = () => {
    mutate({
      nickname,
      realname,
    });
  };
  return (
    <Card>
      <CardContent>
        <Grid container direction="column">
          <Grid item>
            <TextField
              helperText="Name"
              value={realname}
              onChange={(event) => setRealname(event.target.value)}
              label="Name"
            />
          </Grid>
          <Grid item>
            <TextField
              helperText="Nickname"
              value={nickname}
              onChange={(event) => setNickname(event.target.value)}
              label="Nickname"
            />
          </Grid>
        </Grid>
      </CardContent>
      <CardActions>
        <Button
          sx={{
            margin: (theme) => theme.spacing(),
          }}
          variant="outlined"
          onClick={onCreatePlayer}
        >
          Create
        </Button>
      </CardActions>
    </Card>
  );
};

type PlayersGridProps = {
  tournament: string;
};

const PlayersGrid = ({ tournament }: PlayersGridProps) => {
  const { status, error, data } = useTournamentPlayers(tournament);
  if (status === 'loading') {
    return <CircularProgress />;
  }
  if (status === 'error') {
    return <Error msg={error?.message}></Error>;
  }
  return (
    <Grid container spacing={16} direction="row">
      {data?.map((player, _) => (
        <Grid item key={player.nickname}>
          <Player player={player} tournament={tournament} />
        </Grid>
      ))}
      <Grid item>
        <NewPlayer />
      </Grid>
    </Grid>
  );
};

export default PlayersGrid;
