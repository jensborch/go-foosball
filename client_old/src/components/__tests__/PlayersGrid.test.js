import React from 'react';
import PlayersGrid from '../PlayersGrid';
import renderer from 'react-test-renderer';

describe('PlayersGrid', () => {
  const props = {
    tournament: 'uuid',
    players: [
      {
        nickname: 'nick',
        realname: 'name',
      },
    ],
    ranking: 42,
    classes: {},
    deselect: () => {},
    select: () => {},
    create: () => {},
    score: 1500,
  };
  test('Should render players', () => {
    const component = renderer.create(<PlayersGrid {...props} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
