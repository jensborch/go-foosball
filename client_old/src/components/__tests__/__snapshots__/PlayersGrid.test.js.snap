// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayersGrid Should render players 1`] = `
<div
  className="MuiGrid-container-1 MuiGrid-spacing-xs-16-24"
>
  <div
    className="MuiGrid-item-2"
  >
    <div
      className="MuiPaper-root-99 MuiPaper-elevation1-102 MuiPaper-rounded-100 MuiCard-root-98"
    >
      <div />
      <div
        className="MuiCardContent-root-126"
      >
        <h3
          className="MuiTypography-root-127 MuiTypography-h5-143 MuiTypography-gutterBottom-154"
        >
          nick
           - 
          name
        </h3>
      </div>
      <div
        className="MuiCardActions-root-163"
      >
        <div
          className="MuiCardActions-action-165"
        >
          <div
            className="MuiFormControl-root-166 MuiFormControl-marginDense-168"
            lable="Ranking"
          >
            <div
              className="MuiInputBase-root-183 MuiInput-root-170 MuiInput-underline-174 MuiInputBase-formControl-184 MuiInput-formControl-171 MuiInputBase-marginDense-190"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                className="MuiInputBase-input-193 MuiInput-input-178 MuiInputBase-inputType-196 MuiInput-inputType-181 MuiInputBase-inputMarginDense-194 MuiInput-inputMarginDense-179"
                disabled={false}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="number"
                value={42}
              />
            </div>
            <p
              className="MuiFormHelperText-root-200 MuiFormHelperText-marginDense-203 MuiFormHelperText-filled-206"
            >
              Ranking
            </p>
          </div>
          <button
            className="MuiButtonBase-root-234 MuiButton-root-208 MuiButton-outlined-216"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiButton-label-209"
            >
              Add
            </span>
            <span
              className="MuiTouchRipple-root-237"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-item-2"
  >
    <div
      className="MuiPaper-root-99 MuiPaper-elevation1-102 MuiPaper-rounded-100 MuiCard-root-98"
    >
      <div
        className="MuiCardContent-root-126"
      >
        <div
          className="MuiGrid-container-1 MuiGrid-direction-xs-column-4"
        >
          <div
            className="MuiGrid-item-2"
          >
            <div
              className="MuiFormControl-root-166"
              lable="Name"
            >
              <div
                className="MuiInputBase-root-183 MuiInput-root-170 MuiInput-underline-174 MuiInputBase-formControl-184 MuiInput-formControl-171"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  className="MuiInputBase-input-193 MuiInput-input-178"
                  disabled={false}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                  value=""
                />
              </div>
              <p
                className="MuiFormHelperText-root-200"
              >
                Name
              </p>
            </div>
          </div>
          <div
            className="MuiGrid-item-2"
          >
            <div
              className="MuiFormControl-root-166"
              lable="Nickname"
            >
              <div
                className="MuiInputBase-root-183 MuiInput-root-170 MuiInput-underline-174 MuiInputBase-formControl-184 MuiInput-formControl-171"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  className="MuiInputBase-input-193 MuiInput-input-178"
                  disabled={false}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                  value=""
                />
              </div>
              <p
                className="MuiFormHelperText-root-200"
              >
                Nickname
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiCardActions-root-163"
      >
        <button
          className="MuiButtonBase-root-234 MuiButton-root-208 MuiButton-outlined-216 MuiCardActions-action-165"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-209"
          >
            Create
          </span>
          <span
            className="MuiTouchRipple-root-237"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
